### PR TITLE
BrowserRouter

### PR DESCRIPTION
**Pull request by sanyavertolet**
It _might_ work

### What's done:
 * Enriched webpack configuration in order to support `BrowserRouter`
 * Replaced all the relative paths to resources to absolute ones
 * Removed all the logic connected with smart urls (each tab used to have its own url)
 * Replaced all the `Location.hash` occurrences with `Location.pathname`
 * Tested locally with and without `gateway` on

### CODE REVIEW

File: api-gateway/src/main/resources/error.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
     <title>SAVE Cloud Service</title>
 
-    <link rel="stylesheet" type="text/css" href="main.css">
+    <link rel="stylesheet" type="text/css" href="/main.css">
```
Comment (akuleshov7) [inline api-gateway/src/main/resources/error.html:13]: Ah, that was the reason, lol

**Review by akuleshov7 — APPROVED** [review]
Let’s test on prod, but fix tests…

File: api-gateway/src/main/resources/error.html (hunk 1 above)
Comment (sanyavertolet) [reply api-gateway/src/main/resources/error.html:13]: The main reason for a problem was some plugin for HTML processing. It used to add a script tat to the head of html. The path was relative there.

**Comment by sanyavertolet** [comment]
> but fix tests…

Seems that `AppTest` is not `App` test anymore. It is `IndexTest`. `BrowserRouter` seem to work wrong now, as within test it gives `Not Found` page while running the frontend locally works as expected: `indexView` is rendered.

**Comment by akuleshov7** [comment]
> > but fix tests…
> 
> Seems that `AppTest` is not `App` test anymore. It is `IndexTest`. `BrowserRouter` seem to work wrong now, as within test it gives `Not Found` page while running the frontend locally works as expected: `indexView` is rendered.

Fuck tests then, delete them

**Pull request merged**